paparazzi/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -59,10 +59,14 @@ class PaparazziPlugin : Plugin<Project> {
       }
     }
 
-    project.plugins.withId("com.android.library") { setupPaparazzi(project) }
+    val extension = project.extensions.create("paparazzi", PaparazziPluginExtension::class.java)
+    extension.snapshotDirectory.convention(project.layout.projectDirectory.dir("src/test/snapshots"))
+    project.plugins.withId("com.android.library") {
```

Comment:
Why is it necessary to register the extension outside of where it's actually used. If the library plugin is not applied, this create is redundant.

Reply:
> I would expect the project compiles

ah, do you mean that build.gradle.kts wouldn't work because the extension is not available without the lib plugin?

Reply:
Exactly.

Reply:
makes sense, resolve (I'm saying resolve because I don't see the button 🤔)